################################################################################
SHORT DESCRIPTION: 
################################################################################
QSB-023 (https://github.com/QubesOS/qubes-secpack/blob/master/QSBs/qsb-023-2015)
XSA-155 additional patches for the Xen network and block frontends from Qubes
OS Project.

################################################################################
LONG DESCRIPTION: 
################################################################################
Source: https://github.com/QubesOS/qubes-linux-kernel/tree/stable-3.18/patches.xen
Patch: xsa155-linux-0011-xen-netfront-add-range-check-for-Tx-response-id.patch

Tx response ID is fetched from shared page, so make sure it is sane
before using it as an array index.

################################################################################
PATCHES 
################################################################################
Index: linux-4.4/drivers/net/xen-netfront.c
===================================================================
--- linux-4.4.orig/drivers/net/xen-netfront.c	2016-01-25 17:39:55.973741932 +0100
+++ linux-4.4/drivers/net/xen-netfront.c	2016-02-10 18:49:18.861539476 +0100
@@ -379,6 +379,7 @@
 				continue;
 
 			id  = txrsp->id;
+			BUG_ON(id >= NET_TX_RING_SIZE);
 			skb = queue->tx_skbs[id].skb;
 			if (unlikely(gnttab_query_foreign_access(
 				queue->grant_tx_ref[id]) != 0)) {
